src/Dmishh/Bundle/SettingsBundle/DependencyInjection/Configuration.php

```diff
@@ -32,8 +32,18 @@ public function getConfigTreeBuilder()
             SettingsManagerInterface::SCOPE_USER,
         );
 
+        $supportedDrivers = array('orm', 'mongodb');
+
         $rootNode
             ->children()
+                ->scalarNode('db_driver')
+                    ->validate()
+                        ->ifNotInArray($supportedDrivers)
+                        ->thenInvalid('The driver %s is not supported. Please choose one of '.json_encode($supportedDrivers))
```

Comment:
This could be `implode` instead of `json_encode` I think? Rest of the code looks quite okay, but I've got no MongoDB experience to validate this.